EC2 Instance and Postman

AmoghaK1 opened the pull request:
Deployed website on EC2 Instance 
Added Postman collection Json file in the Postman folder.
Used MongoDB Atlas to store images and maintaining 100% uptime.


Public IP Address:
http://43.205.212.117:3000/

AmoghaK1 committed:
Deployed on ec2 instance and added postman json

AmoghaK1 committed:
patch fix

AmoghaK1 committed:
image upload fix

cpandey05 commented:
Why are you using port 3000? change to 80 so that it is directly accessible and restrict EC2 security groupt to only HTTP TCP ports

AmoghaK1 commented:
> Why are you using port 3000? change to 80 so that it is directly accessible and restrict EC2 security groupt to only HTTP TCP ports

Updated the port to 80. 
Please Verify. 
http://13.204.43.148/

cpandey05 commented:
looks good, did you moved images to S3, do not forget to follow layered approach .. write service for S3 in reusable way. Also research what are the different ways to connect to S3

AmoghaK1 committed:
S3 Integrated

AmoghaK1 committed:
fixed: mobile uploads

AmoghaK1 committed:
bug fix: mobile uploads

AmoghaK1 commented:
> looks good, did you moved images to S3, do not forget to follow layered approach .. write service for S3 in reusable way. Also research what are the different ways to connect to S3

Update:
Images are now uploaded to S3. Metadata and reviews are still stored in MongoDB. 
Currently connected to S3 using Multer. 
Also added a lifecycle rule which deletes images older than 2 months.

Please can you ellaborate, "write service for S3 in reusable way". What is expected?

cpandey05 commented on services/mainService.js:
```diff
@@ -8,28 +8,42 @@ const ReviewDTO = require('../dtos/ReviewDTO');
+    };
   } catch (error) {
     console.error('Error in uploadImage:', error);
     throw error;
```
What user will see on screen if an error occurs, what if i want to retry?

AmoghaK1 replied:
![WhatsApp Image 2025-05-31 at 16 51 23](https://github.com/user-attachments/assets/5172f107-926b-42fc-91b0-b7d18df28d0e)

currently errors are shown this way

cpandey05 commented:
Have you added server and client in same project? the layering does not look clean

AmoghaK1 commented:
> Have you added server and client in same project? the layering does not look clean

I have used basic html css for frontend which is in the views folder. 
I only have 2 files for the client side so didnt create a client repo

AmoghaK1 committed:
modified:   routes/mainRoutes.js
	modified:   services/mainService.js

AmoghaK1 committed:
better error messages

AmoghaK1 committed:
S3 Bucket changed to Private

AmoghaK1 committed:
patch fix

AmoghaK1 committed:
Synced S3 and MongoDB - Delete Images

AmoghaK1 commented:
Changed S3 Bucket to Private. 
Images are fetched using temporary signed URLs. 

Added lifecycle rule which deletes images from S3 after 5 days. 
Synced MongoDB and S3, hence when images now get deleted from S3, Metadata also gets deleted from MongoDB 
Achieved this using "node cron", which runs a script every day. The script fetches objects from S3 and compares them with Metadata, deleting any unmatched images.

AmoghaK1 committed:
MIGRATED TO DYNAMODB COMPLETELY

AmoghaK1 commented:
Migrated from MongoDB to DynamoDB.
Created a Single Table which includes a Partition and Sort key to identify Metadata and Reviews for images. 
Removed all unused MongoDB files.

AmoghaK1 committed:
Update README.md

AmoghaK1 committed:
modified:   README.md

AmoghaK1 committed:
Merge branch 'deployment/ec2-setup' of https://github.com/AmoghaK1/Real-Review into deployment/ec2-setup

AmoghaK1 committed:
code fix

cpandey05 commented on dynamo-models/imageDynamoModel.js:
```diff
@@ -0,0 +1,37 @@
+const params = {
+  TableName: 'RealReview',
+  KeySchema: [
+    { AttributeName: 'PK', KeyType: 'HASH' }, // Partition key
```
What is partition key and sort key

cpandey05 commented on public/home.js:
```diff
@@ -0,0 +1,215 @@
+    fileInput.addEventListener('change', function() {
+      if (this.files && this.files[0]) {
+        fileNameDisplay.textContent = this.files[0].name;
+      } else {
```
Why hard coded index , is there a better way

cpandey05 commented on public/home.js:
```diff
@@ -0,0 +1,215 @@
+// Handle file selection
```
Try to have a separate UI project

cpandey05 commented:
Your branch have conficts, resolve it so that it can be merged
